File: busca-binaria/busca_binaria.js

function buscaBinaria(arr, item) {
  let low = 0
  let high = arr.length - 1

  while (low <= high) {
    let middle = (low + high) / 2
    let value = arr[middle]

    if (value === item) {
      return middle
    }
    if (value > item) {
      middle ++
    }
    else {
      low = middle + 1
    }
  }
  return null
}

const arr = [
  1,3,5,7,9
]

console.log(buscaBinaria(arr, 3))

File: busca-binaria/busca_binaria.swift

func buscaBinaria(arr: [Int], number: Int) -> Int {
    let baixo: Int = 0
    let meio: Int = (arr.count / 2) - 1
    let maior: Int = arr.count - 1

    while baixo <= maior {
        if number == arr[meio] {
            return meio
            break
        }
    }

    return 0
}

let arr = [2, 5, 6, 8, 10, 45, 80]

print(buscaBinaria(arr: arr, number: 8))
